### PR TITLE
loot tracker: show HA prices

### PR DESCRIPTION
**Pull request by Hydrox6**
Closes: #5813
Closes: #7858

Adds an option to use HA values in the loot tracker, along with an option to visually show what type each price is. Also adds both prices to the item tooltips.

tooltip
GE prices with type shown
HA prices with type shown

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java [hunk 1]
```diff
@@ -49,6 +49,26 @@ default String getIgnoredItems()
...
+	{
+		return false;
+	}
+
+	@ConfigItem(
+		keyName = "showPriceType",
+		name = "Show Price Type",
+		description = "Whether to show a GE: or HA: next to the total values in the tracker"
+	)
+	default boolean showPriceType()
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java]: I dont really think this is necessary

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerBox.java [hunk 2]
```diff
@@ -78,13 +79,15 @@
 	private BiConsumer<String, Boolean> onItemToggle;
 
 	LootTrackerBox(
+		final LootTrackerConfig config,
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerBox.java]: send config option and not whole config

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPanel.java [hunk 3]
```diff
@@ -622,8 +622,14 @@ private void updateOverall()
...
 
+		String priceType = "";
+		if (config.showPriceType())
+		{
+			priceType = config.preferHA() ? "HA " : "GE ";
+		}
+
 		overallKillsLabel.setText(htmlLabel("Total count: ", overallKills));
-		overallGpLabel.setText(htmlLabel("Total value: ", overallGp));
+		overallGpLabel.setText(htmlLabel("Total " + priceType + "value: ", overallGp));
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPanel.java]: This should have tooltip with alt price type

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java (hunk 1 above)
Comment (Hydrox6) [reply runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java]: I made this an option so people who didn't want it don't have to use it.

**Comment by Hydrox6** [comment]
Updated with feedback, including @raiyni's, which didn't go through due to mobile. Still not entirely sure if a button in the panel would be required.

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java (hunk 1 above)
Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java]: Well it is not adding any visual clutter and there is plenty space on loot tracker so its pretty pointless, but sure I guess

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java [hunk 4]
```diff
@@ -49,6 +49,16 @@ default String getIgnoredItems()
...
 	void setIgnoredItems(String key);
 
+	@ConfigItem(
+		keyName = "priceType",
+		name = "Price Type",
+		description = "Show High Alchemy prices for the box total (both prices are still available in the tooltip)"
+	)
+	default LootTrackerPriceType priceType()
+	{
+		return LootTrackerPriceType.GRANGE_EXCHANGE;
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java]: grange?

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java [hunk 5]
```diff
@@ -49,6 +49,16 @@ default String getIgnoredItems()
 	)
 	void setIgnoredItems(String key);
 
+	@ConfigItem(
+		keyName = "priceType",
+		name = "Price Type",
+		description = "Show High Alchemy prices for the box total (both prices are still available in the tooltip)"
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java]: This looks left over from when it was a bool?

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java [hunk 6]
```diff
@@ -49,6 +49,26 @@ default String getIgnoredItems()
...
+	{
+		return LootTrackerPriceType.GRANGE_EXCHANGE;
+	}
+
+	@ConfigItem(
+		keyName = "showPriceType",
+		name = "Show Price Type",
+		description = "Whether to show a GE: or HA: next to the total values in the tracker"
+	)
+	default boolean showPriceType()
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java]: I dont see why this should be configurable?

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java (hunk 4 above)
Comment (Hydrox6) [reply runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java]: Whoops, this is what I get for coding when ill. ty

**Pull request merged**